src/components/aside/tools.js

```diff
@@ -0,0 +1,99 @@
+import BookmarkIcon from './assets/tool-bookmark.svg'
+import FBIcon from './assets/share-fb.svg'
+import LineIcon from './assets/share-line.svg'
+import TwitterIcon from './assets/share-twitter.svg'
+import PrintIcon from './assets/tool-print.svg'
+import PropTypes from 'prop-types'
+import React from 'react'
+import TextIcon from './assets/tool-text.svg'
+import predefinedProps from './prop-types'
+import styled from 'styled-components'
+
+const ToolsBlock = styled.div`
+  width: 20px;
+  height: ${props => props.height};
+  display: flex;
+  flex-direction: column;
+  justify-content: space-around;
+  align-items: center;
+
+  > svg {
+    cursor: pointer;
+  }
+`
+
+function FBShareBT(props) {
+  const handleClick = () => {
+    const currentURL = window.location.href
+    const location =
+      'https://www.facebook.com/dialog/feed?' +
+      'display=page' +
+      `&app_id=${props.appID}` +
+      `&link=${encodeURIComponent(currentURL)}` +
+      `&redirect_uri=${encodeURIComponent('https://www.facebook.com/')}`
+
+    window.open(location, '_blank')
+  }
+
+  return <FBIcon onClick={handleClick} />
+}
+
+FBShareBT.propTypes = {
+  appID: PropTypes.string.isRequired,
+}
+
+function TwitterShareBT(props) {
+  const handleClick = () => {
+    const currentURL = window.location.href
+    const location =
+      'https://twitter.com/intent/tweet?' +
+      `text=${props.title} ${encodeURIComponent(currentURL)}`
```

Comment:
I'll prefer to use the query parameter of Twitter web intent. And Twitter will append a preview of url to the post, including the `title` of article, so I think we don't have to add title to the text (especially when Twitter has limit of characters for each post)

```js
    const location =
      'https://twitter.com/intent/tweet?' +
      `url=${encodeURIComponent(currentURL)}`
```

Ref: https://developer.twitter.com/en/docs/twitter-for-websites/tweet-button/guides/web-intent